src/SprykerEco/Zed/AmazonQuicksight/Business/Deleter/QuicksightUserDeleter.php

```diff
@@ -0,0 +1,231 @@
+        if ($quicksightListUsersResponseTransfer->getErrors()->count() !== 0) {
+            return $quicksightUserCollectionResponseTransfer->setErrors(
+                $quicksightListUsersResponseTransfer->getErrors(),
+            );
+        }
+
+        $quicksightUserTransfers = $quicksightListUsersResponseTransfer->getQuicksightUsers();
+        if ($quicksightUserTransfers->count() === 0) {
+            return $quicksightUserCollectionResponseTransfer;
+        }
+
+        $notMatchedQuicksightUserTransfers = $this->quicksightUserMatcher->getQuicksightUsersNotMatchedWithExistingUsers(
+            $quicksightUserTransfers,
+        );
+
+        foreach ($notMatchedQuicksightUserTransfers as $quicksightUserTransfer) {
+            $quicksightUserCollectionResponseTransfer = $this->deleteQuicksightUser(
+                $quicksightUserTransfer,
+                $quicksightUserCollectionResponseTransfer,
+            );
+        }
+
+        return $quicksightUserCollectionResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UserCollectionResponseTransfer $userCollectionResponseTransfer
+     * @param array<int|string, \Generated\Shared\Transfer\UserTransfer> $userTransfers
+     *
+     * @return \Generated\Shared\Transfer\UserCollectionResponseTransfer
+     */
+    protected function executeDeleteQuicksightUsersByUserCollectionResponseTransaction(
+        UserCollectionResponseTransfer $userCollectionResponseTransfer,
+        array $userTransfers
+    ): UserCollectionResponseTransfer {
+        $userIdsToDelete = [];
+        foreach ($userTransfers as $entityIdentifier => $userTransfer) {
+            $quicksightDeleteUserResponseTransfer = $this->amazonQuicksightApiClient->deleteUserByUsername($userTransfer);
+            if ($quicksightDeleteUserResponseTransfer->getErrors()->count() !== 0) {
+                $userCollectionResponseTransfer = $this->addErrorsToUserCollectionResponse(
+                    $userCollectionResponseTransfer,
+                    $quicksightDeleteUserResponseTransfer->getErrors(),
+                    (string)$entityIdentifier,
+                );
+
+                $this->messengerFacade->addInfoMessage(
+                    (new MessageTransfer())->setValue(static::ERROR_MESSAGE_QUICKSIGHT_USER_DELETION_FAILED),
+                );
+
+                continue;
+            }
+
+            $userIdsToDelete[] = $userTransfer->getIdUserOrFail();
+        }
+
+        $this->amazonQuicksightEntityManager->deleteQuicksightUsersByUserIds($userIdsToDelete);
+
+        return $userCollectionResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightUserTransfer $quicksightUserTransfer
+     * @param \Generated\Shared\Transfer\QuicksightUserCollectionResponseTransfer $quicksightUserCollectionResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuicksightUserCollectionResponseTransfer
+     */
+    protected function deleteQuicksightUser(
```

Comment:
* Do we need to remove the quicksight user from the local DB?

Reply:
Not in this case, it is used in console command which removes users from QS only